credentials/src/main/kotlin/web5/sdk/credentials/PresentationExchange.kt

```diff
@@ -136,6 +137,26 @@ public object PresentationExchange {
     PresentationDefinitionV2Validator.validate(presentationDefinition)
   }
 
+  /**
+   * Validates whether an object is usable as a presentation submission or not.
+   *
+   * Model as specified in https://identity.foundation/presentation-exchange/#presentation-submission.
+   *
+   * The checks are as follows:
+   * 1. Ensures that the presentation submission's id is not empty.
+   * 2. Validates that the definitionId is not empty.
+   * 3. Validates descriptorMap is a non-empty list.
+   * 4. Check for unique inputDescriptor ids at top level
+   * 5. Verifies the input descriptor mapping ids are the same on all levels of nesting.
+   * 6. Ensures that the path is valid across all levels of nesting
+   *
+   * Throws an [PexValidationException] if the provided object does not conform to the Presentation Definition
+   */
+  @Throws(PexValidationException::class)
+  public fun validateSubmission(presentationSubmission: PresentationSubmission) {
```

Comment:
this seems to differ from the JS version which returns a result vs throwing an exception

```js
const submissionCheck = PresentationExchange.validateSubmission({
    presentationSubmission: presentationResult.presentationSubmission
  });
  ```
  
  Any reason why?

Reply:
So actually the initial thought on all the verifies was to throw an exception if it failed, and not have the results object.

In JS I happened to use the raw underlying Results object instead of wrapping it with an exception (maybe a mistake)

To keep it consistent I thinik we should decide if we want to use a `Results` object everywhere, OR if we want to 'fix' the JS and remove the existing results object and go back to throwing an exception

Tracking the discussion on that decision here - https://github.com/TBD54566975/web5-js/issues/386